### ISSUE TITLE
Allow JIRA regex matching to match multiple issues

### ISSUE THREAD

**Issue by bbaetz** [mohamicorp/yet-another-commit-checker#15]
When the regex matcher isn't configured, YACC checks that all the issues in the commit message are valid. However, when a regex matcher is configured, only the first match is checked.

The regex matcher should be able to be run multiple times on the message to find and check all of the issues.

**Comment by christiangalsterer** [comment]
Hi,

any news on this? What about an additional config switch which enables either multiline matches or exact match to preserve backwards compatibility?

Br
Christian

**Comment by sford** [comment]
I think there are other issues besides preserving backwards compatibility. @bbaetz proposed a potential solution that involved a new JIRA extractor regex that looked attractive because it doesn't touch the existing commit regex. I looked into what it would take to implement that after closing #47, but kepted on running into weird corner cases that didn't feel right, mostly due to the use of regex `find`.

For example, I added a new regex that allowed you to add an extractor regex to pull JIRA issues using `find`, something like `\[[A-Z-\d]+\]`. This allows you to accurately pull JIRA issues from your example, but that regex would skip any sort of typo that causes the regex to not match, like `[ABC123]`, `ABC-123:` `[ABC-123 fixed issue`, `ABC-123 fixed issue`.

You could make the regex really relaxed, like `[A-Z]+-\d+`, but then it would operate not much differently than if you just clicked `Require Valid JIRA Issue` and maybe `Ignore Unknown JIRA Project Keys` to prevent false positives.

To solve the typo problem, you can then use the commit regex to first ensure the commit message matches this exactly before extracting issues:

    [ABC-123] fix one
    [ABC-123] fix two

A regex might look like: `(\[[A-Z-\d]+\] .*\n?)+` (untested, but close I think)

If someone wants to instead support a multi-line version of this, like:

    [ABC-123] fix issue
        more info

    [BC-333] another issue
        more info

They could enable DOTALL using `(?s)` to accommodate this, but this would cause `.*` to match the whole commit message, which means all subsequent `[issueId]` won't be matched, allow
ing typos.

tl;dr: Enforcing structure *and* matching multiple issues is difficult. It works in some isolated cases, but not others, which introduces pitfalls that are easy to fall into. Maybe
the answer is back towards a single regex (e193b41), but I don't fully understand the nuances and worry about making full structure difficult to enforce or introducing corner cases.
